fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsApplicationProcessWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -446,6 +446,72 @@ public CommandProcessingResult rejectApplication(final Long savingsId, final Jso
                 .build();
     }
 
+	@Transactional
+    @Override
+    public CommandProcessingResult undoApplicationRejection(final Long savingsId, final JsonCommand command) {
+        this.context.authenticatedUser();
+
+        this.savingsAccountApplicationTransitionApiJsonValidator.validateForUndo(command.json());
+
+        final SavingsAccount savingsAccount = this.savingAccountAssembler.assembleFrom(savingsId);
+        checkClientOrGroupActive(savingsAccount);
+
+        final Map<String, Object> changes = savingsAccount.undoApplicationRejection();
+        if (!changes.isEmpty()) {
+            this.savingAccountRepository.save(savingsAccount);
+
+            final String noteText = command.stringValueOfParameterNamed("note");
+            if (StringUtils.isNotBlank(noteText)) {
+                final Note note = Note.savingNote(savingsAccount, noteText);
+                changes.put("note", noteText);
+                this.noteRepository.save(note);
+            }
+        }
+
+        return new CommandProcessingResultBuilder() //
+                .withCommandId(command.commandId()) //
+                .withEntityId(savingsId) //
+                .withOfficeId(savingsAccount.officeId()) //
+                .withClientId(savingsAccount.clientId()) //
+                .withGroupId(savingsAccount.groupId()) //
+                .withSavingsId(savingsId) //
+                .with(changes) //
+                .build();
+    }
+
+    @Transactional
+    @Override
+    public CommandProcessingResult reopen(final Long savingsId, final JsonCommand command) {
```

Comment:
@nazeer1100126 , Actually that depends on what payment method has been used. For example, if the payment method is cash during account closure, then when the account is reopened the balance is zero as this directly affected cash. If payment method is not by cash (i.e by Voucher) then the balances at closure are to be booked to a GL account say "Account Closure Account" and when the account is reopened, the balance is got from the GL Account (Account Closure Account) and posted back to the account reopened.